pkg/universe.dagger.io/x/ezequiel@foncubierta.com/terraform/run.cue

```diff
@@ -0,0 +1,92 @@
+package terraform
+
+import (
+  "dagger.io/dagger"
+  "dagger.io/dagger/core"
+  "universe.dagger.io/docker"
+)
+
+// Terraform log level
+#LogLevel: "off" | "info" | "warn" | "error" | "debug" | "trace"
+
+// Default log level
+_#DefaultLogLevel: "off"
+
+// Run `terraform CMD`
+#Run: {
+  // Terraform source code
+  source: dagger.#FS
+
+  // Terraform command (i.e. init, plan, apply)
+  cmd: string
+
+  // Arguments for the Terraform command (i.e. -var-file, -var)
+  cmdArgs: [...string]
+
+  // Arguments set internally from other actions
+  withinCmdArgs: [...string]
+
+  // Terraform workspace
+  workspace: string | *"default"
+
+  // Data directory (i.e. ./.terraform)
+  dataDir: string | *".terraform"
+
+  // Log level
+  logLevel: #LogLevel | *_#DefaultLogLevel
+
+  // Log path
+  logPath: string | *""
+
+  // Environment variables
+  env: [string]: string | dagger.#Secret
+
+  // Environment variables set internally from other actions
+  withinEnv: [string]: string | dagger.#Secret
+
+  // Joins user and internal command arguments
+  _thisCmdArgs: cmdArgs + withinCmdArgs
+
+  // Joins user and internal environment variables
+  _thisEnv: env & withinEnv & {
+    TF_WORKSPACE: workspace
+    TF_DATA_DIR: dataDir
+    TF_LOG: logLevel
+    TF_LOG_PATH: logPath
+    TF_IN_AUTOMATION: "true"
+    TF_INPUT: "0"
+  }
+
+  // Hashicorp Terraform container
+  container: #input: docker.#Image | #Image
+
+  // Run command within a container
+  _run: docker.#Build & {
+    steps: [
+      container.#input,
+      docker.#Copy & {
+        dest: "/src"
+        contents: source
+      },
+      docker.#Run & {
+        workdir: "/src"
+        command: {
+          name: cmd
+          args: _thisCmdArgs
+        }
+        env: _thisEnv
+      }
+    ]
+  }
```

Comment:
You could directly use `docker.#Run` outside `docker.#Build`, I'll try your package and show you ;)

Reply:
For completeness, this is what I tried and didn't work

```
#Run: {
  ...

  _image: #Image

  // Run command within a container
  _run: docker.#Run & {
    input: *_image.output | docker.#Image
    workdir: "/src"
    command: {
      name: cmd
      args: _thisCmdArgs
    }
    mounts: src: {
      dest: "/src"
      contents: source
    }
    env: _thisEnv
  }

  ...
}
```

Reply:
Oh you are right, you are updating `/src` and outputting it for further operation so yes, you did great!